src/test/java/io/github/bobocodebreskul/context/registry/SimpleBeanDefinitionRegistryTest.java

```diff
@@ -0,0 +1,105 @@
+package io.github.bobocodebreskul.context.registry;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
+
+import io.github.bobocodebreskul.context.config.BeanDefinition;
+import io.github.bobocodebreskul.context.exception.BeanDefinitionDuplicateException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+class SimpleBeanDefinitionRegistryTest {
+
+  private BeanDefinitionRegistry beanDefinitionRegistry;
+
+  @BeforeEach
+  void init() {
+    beanDefinitionRegistry = new SimpleBeanDefinitionRegistry();
+  }
+
+  @Test
+  void givenBeanNameAndBeanDefinitionWhenBeanDefinitionNotRegisterThenRegisterBeanDefinition() {
+    BeanDefinition beanDefinitionMock1 = Mockito.mock(BeanDefinition.class);
+    beanDefinitionRegistry.registerBeanDefinition("testBeanName1", beanDefinitionMock1);
+    assertSame(beanDefinitionMock1, beanDefinitionRegistry.getBeanDefinition("testBeanName1"));
+  }
+
+  @Test()
+  void givenBeanNameAndBeanDefinitionWhenBeanDefinitionRegisterDuplicateThenThrow() {
+    BeanDefinition beanDefinitionMock1 = Mockito.mock(BeanDefinition.class);
+    BeanDefinition beanDefinitionMock2 = Mockito.mock(BeanDefinition.class);
+    String testBeanName = "testBeanName";
+    beanDefinitionRegistry.registerBeanDefinition(testBeanName, beanDefinitionMock1);
+    Exception exception = assertThrows(
+        BeanDefinitionDuplicateException.class,
+        () -> beanDefinitionRegistry.registerBeanDefinition(testBeanName, beanDefinitionMock2));
+
+    String expectedMessage =
+        "Cannot registered a beanDefinition with name %s because a beanDefinition with that name is already registered%n".formatted(
+            testBeanName);
+    String actualMessage = exception.getMessage();
+
+    assertTrue(actualMessage.contains(expectedMessage));
+  }
+
+  @Test
+  void givenBeanNameThenRemoveBeanDefinition() {
+    BeanDefinition beanDefinitionMock1 = Mockito.mock(BeanDefinition.class);
+    BeanDefinition beanDefinitionMock2 = Mockito.mock(BeanDefinition.class);
+    beanDefinitionRegistry.registerBeanDefinition("testBeanName1", beanDefinitionMock1);
+    beanDefinitionRegistry.registerBeanDefinition("testBeanName2", beanDefinitionMock2);
+
+    assertNotNull(beanDefinitionRegistry.getBeanDefinition("testBeanName1"));
+    beanDefinitionRegistry.removeBeanDefinition("testBeanName");
```

Comment:
What is the point of this action here? If you are testing behavior on unexisting bean name removal, then I assume it's better to do as separate test.